File: library-api/src/main/java/com/library/libraryapi/dto/auth/AuthResponse.java
package com.library.libraryapi.dto.auth;

import com.library.libraryapi.model.User;

public record AuthResponse(
        String token,
        String tokenType,
        UserResponse user
) {
    public static AuthResponse of(String token, User user) {
        return new AuthResponse(token, "Bearer", UserResponse.from(user));
    }
}

File: library-flutter/lib/features/auth/data/models/auth_response.dart
import 'user_response.dart';

class AuthResponse {
  final String token;
  final String tokenType;
  final UserResponse user;

  AuthResponse({
    required this.token,
    required this.tokenType,
    required this.user,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) => AuthResponse(
        token: json['token'] as String,
        tokenType: json['tokenType'] as String? ?? 'Bearer',
        user: UserResponse.fromJson(json['user'] as Map<String, dynamic>),
      );
}
